diff --git a/servers/vfs/protect.c b/servers/vfs/protect.c
index 70a4536..1557aed 100644
--- a/servers/vfs/protect.c
+++ b/servers/vfs/protect.c
@@ -65,6 +65,8 @@ int do_chmod()
 	dup_vnode(vp);
   }
 
+  printf("chmod: %s %o\n", fullpath, new_mode);
+
   /* Only the owner or the super_user may change the mode of a file.
    * No one may change the mode of a file on a read-only file system.
    */
