diff --git a/servers/pm/table.c b/servers/pm/table.c
index 1f4dbbb..677abad 100644
--- a/servers/pm/table.c
+++ b/servers/pm/table.c
@@ -81,7 +81,7 @@ int (*call_vec[])(void) = {
        do_getmcontext, /* 67 = getmcontext */
        do_setmcontext, /* 68 = setmcontext */
        do_getproccount,                /* 69 = getproccount    */
-       no_sys,         /* 70 = unused  */
+       do_getppid,             /* 70 = getppid */
        do_sigaction,   /* 71 = sigaction   */
        do_sigsuspend,  /* 72 = sigsuspend  */
        do_sigpending,  /* 73 = sigpending  */